### PR TITLE
When no gui event loop is running, propagate callback exceptions.

### PR DESCRIPTION
**Pull request by anntzer**
CallbackRegistry currently defaults to suppressing exceptions that occur
in callbacks (since #9063) because they cause PyQt5 to abort() the program with no
chance of even catching the exception, but such a behavior is annoying
when writing tests involving callbacks (e.g., mplcursors), because in
that case we actually want to see whether an exception occurred, and
typically fail the test in that case.

Instead we can detect whether a GUI event loop is currently running and
propagate the exception if none is running.

Part of the patch is just moving `_get_running_interactive_framework`
from `backends` to `cbook`, so that `cbook` keeps its property of being
importable independent of the rest of mpl.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by anntzer** [comment]
rebased

File: doc/api/next_api_changes/2018-09-15-AL.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+API changes
+```````````
```
Comment (timhoffm) [inline doc/api/next_api_changes/2018-09-15-AL.rst]: Please adapt to the new API change doc structure.

Comment (anntzer) [reply doc/api/next_api_changes/2018-09-15-AL.rst]: fixed

**Comment by tacaswell** [comment]
The drop in test coverage was due to there being a dead branch in the test code now that the default case is now in it's own test.

**Pull request merged**